lib/govuk_app_config/govuk_healthcheck/checkup.rb

```diff
@@ -0,0 +1,46 @@
+module GovukHealthcheck
+  STATUSES = [
+    OK = "ok".freeze,
+    WARNING = "warning".freeze,
+    CRITICAL = "critical".freeze,
+  ].freeze
+
+  class Checkup
+    # @param checks [Array] Array of objects/classes that respond to `run`
+    def initialize(checks)
+      @checks = checks
+    end
+
+    def run
+      {
+        status: worst_status,
+        checks: component_statuses,
+      }
+    end
+
+  private
+
+    def component_statuses
+      @component_statuses ||= @checks.each_with_object({}) do |check, hash|
+        check_result = check.details.merge(status: check.status)
```

Comment:
Could we make `check.details` optional, like `check.message`?

Reply:
Sure

Reply:
Updated.